### PR TITLE
Refactor test_batch_normalization

### PR DESCRIPTION
**Pull request by niboshi**
Refactored `test_batch_normalization` to use `testing.backend.inject_backend_tests`.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/normalization_tests/test_batch_normalization.py [hunk 1]
```diff
@@ -10,9 +10,16 @@
 from chainer import gradient_check
 from chainer import testing
 from chainer.testing import attr
+from chainer.testing import backend
 
 
-def _batch_normalization(expander, gamma, beta, x, mean, var):
+def _to_noncontiguous(arrays):
```
[the remark this answers is not in the document]
Comment (niboshi) [reply tests/chainer_tests/functions_tests/normalization_tests/test_batch_normalization.py]: I agree. `_to_fcontiguous`?

[the remark this answers is not in the document]
Comment (kmaehashi) [reply tests/chainer_tests/functions_tests/normalization_tests/test_batch_normalization.py]: Sounds good to me.

[the remark this answers is not in the document]
Comment (niboshi) [reply tests/chainer_tests/functions_tests/normalization_tests/test_batch_normalization.py]: Fixed!

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**